<launch>
  <arg name="camera_name" default="/rrbot/camera1" />
  <arg name="launch_prefix" default="" /> <!-- set to value="gdbserver localhost:10000" for remote debugging -->
  <arg name="node_namespace" default="apriltags2_ros_continuous_node" />
  <arg name="camera_frame" default="camera::camera_link" />
  <arg name="image_topic" default="image_raw" />

  <node pkg="tf" type="static_transform_publisher" name="camera_tf_publisher_debuggg" args="0.0 0.0 0.0 0 0 -0.711 0.704 local_origin $(arg camera_frame) 200"/>

  <!-- Set parameters -->
  <rosparam command="load" file="$(find apriltags2_ros)/config/settings.yaml" ns="$(arg node_namespace)" />
  <rosparam command="load" file="$(find apriltags2_ros)/config/tags_mania.yaml" ns="$(arg node_namespace)" />
  
  <node pkg="apriltags2_ros" type="apriltags2_ros_continuous_node" name="$(arg node_namespace)" clear_params="true" output="screen" launch-prefix="$(arg launch_prefix)" >
    <!-- Remap topics from those used in code to those on the ROS network -->
    <remap from="image_rect" to="$(arg camera_name)/$(arg image_topic)" />
    <remap from="camera_info" to="$(arg camera_name)/camera_info" />

    <param name="camera_name" type="str" value="$(arg camera_name)" />
    <param name="camera_frame" type="str" value="$(arg camera_frame)" />
    <param name="publish_tag_detections_image" type="bool" value="true" />      <!-- default: false -->
    <param name="publish_tf" type="bool" value="true" />
  </node>
</launch>
